Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1889,6 +1889,27 @@ theorem vecMul_one (v : m → α) : v ᵥ* 1 = v := by
   rw [← diagonal_one, vecMul_diagonal, mul_one]
 #align matrix.vec_mul_one Matrix.vecMul_one
 
+@[simp]
+theorem diagonal_mulVec (x : α) (v : m → α) : (diagonal fun _ => x) *ᵥ v = x • v := by
+  ext; simp [mulVec_diagonal]
+
+@[simp]
+theorem natcast_mulVec (x : ℕ) (v : m → α) : x *ᵥ v = x • v := by
+  change diagonal x *ᵥ v = _
+  ext; simp
+
+@[simp]
+theorem intcast_mulVec [IntCast α] (x : ℤ) (v : m → α) : x *ᵥ v = (x : α) • v := by
+  change diagonal x *ᵥ v = _
+  simp
```

Comment:
The generality here is a little weird; could you move this to the `NonUnitalNonAssocRing` section below so that you can drop the `IntCast` assumption?

Reply:
When I tried moving this theorem to the `NonUnitalNonAssocRing` section (lines 1957), lean rejected the statement, it is not able to coerce `x` to a matrix in `x *ᵥ v`, even when I added the instances: `[Fintype m] [DecidableEq α] [Zero α] [IntCast α]`.

Reply:
I sorry, I meant the `NonAssocRing` section

Reply:
No, they were originally in `NonAssocSemiring` I think?

Reply:
Ah yes, that's right, sorry!